### PR TITLE
feat: highlight browser failures

### LINKED ISSUE #792: 🚀 Feature: Highlight browser failures in red
to make it more obvious without wasting eye-movement effort to look at the count haha. I prefer github.com/visionmedia/mocha-matrix for this reason but it's not a good fit as a default

### PR DESCRIPTION
**Pull request by mark-wiemer**
## PR Checklist

- [x] Addresses an existing open issue: fixes #792, works on #5129
- [x] That issue was marked as [`status: accepting prs`](https://github.com/mochajs/mocha/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/mochajs/mocha/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

- Change pass indicator to accessible color in light mode (dark green)
- As soon as a test fails, add a red fail indicator and color the "failures" text red
- As soon as all tests have a result, and none are failures, add a green pass indicator and color the "passes" text green
- Result indicators for the root suite are the same for the result indicators for individual tests
    - Except result indicator for root suite is only pass/fail, `skip` for root suite will show `pass` for simplicity (happy to change)

image

image

<br />

image

image

### CODE REVIEW

File: mocha.css [hunk 1]
```diff
@@ -36,48 +35,47 @@
...
+    --mocha-code-init: #9cc7f1;
+    --mocha-code-string: #80d4ff;
+    --mocha-code-keyword: #e3a470;
+    --mocha-code-number: #4ca7ff;
   }
 }
 
 body {
-  margin:0;
+  margin: 0;
```
Comment (JoshuaKGoldberg) [inline mocha.css:72]: Could you please revert unrelated changes? I personally prefer them the edited way (were the edits from Prettier? 😄) but it pollutes the diff a bit.

If we want to auto-format this file, it'd be good to do it in a separate standalone issue + PR. There's probably some tooling work to be done to standardize the repo's formatting.

File: lib/reporters/html.js [hunk 2]
```diff
@@ -219,7 +235,6 @@ function HTML(runner, options) {
   }
 
   function updateStats() {
-    // TODO: add to stats
```
Comment (JoshuaKGoldberg) [inline lib/reporters/html.js:222]: [Praise] Nice to see a TODO resolved 😄 I hadn't even noticed this one!

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Looks great to me, thanks for the overhaul! ✨ 

Just requesting undoing unrelated changes, to make it easier to review & not take up as much of a diff.

File: mocha.css (hunk 1 above)
Comment (mark-wiemer) [reply mocha.css:72]: works for me :)

File: lib/reporters/html.js (hunk 2 above)
Comment (mark-wiemer) [reply lib/reporters/html.js:222]: note I just removed this, it looked completed and was from 2011 and it was bothering me :D

**Comment by mark-wiemer** [comment]
@JoshuaKGoldberg, ready for review :)

File: lib/reporters/html.js [hunk 3]
```diff
@@ -127,6 +145,10 @@ function HTML(runner, options) {
 
   runner.on(EVENT_SUITE_END, function (suite) {
     if (suite.root) {
+      if (stats.failures === 0) {
+        text(resultIndicator, '✓');
```
Comment (mark-wiemer) [inline lib/reporters/html.js:149]: Note that this will show a check even if all tests are skipped, I hope that's OK, I see we have a skipped symbol but I didn't know if supporting that at root level was desired / in-scope for this PR, so I left it like this

**Review by JoshuaKGoldberg — APPROVED** [review]
Looks great to me, thanks for iterating on it & coming up with good ideas! 🙌 

Approved - and will leave open for a bit in case anybody from @mochajs/maintenance-crew has time & interest to review.

**Comment by mark-wiemer** [comment]
Thanks @JoshuaKGoldberg, glad I could help. Just curious, is there an ETA for when this might be merged? Would love to see my work deployed (I am a selfish and proud man 😆)
